tests/utils/test_validation.py

```diff
@@ -198,29 +238,64 @@ def test_validate_batch_log_data():
     )
 
 
-def test_validate_experiment_artifact_location():
-    _validate_experiment_artifact_location("abcde")
-    _validate_experiment_artifact_location(None)
-    with pytest.raises(MlflowException, match="Artifact location cannot be a runs:/ URI"):
-        _validate_experiment_artifact_location("runs:/blah/bleh/blergh")
+@pytest.mark.parametrize(
+    ("location", "is_good_location"),
+    [
+        ("abcde", True),
+        (None, True),
+        # error cases
+        ("runs:/blah/bleh/blergh", False),
+    ],
+)
+def test_validate_experiment_artifact_location(location, is_good_location):
+    if is_good_location:
+        _validate_experiment_artifact_location(location)
+    else:
+        with pytest.raises(MlflowException, match="Artifact location cannot be a runs:/ URI"):
+            _validate_experiment_artifact_location(location)
 
 
-def test_validate_experiment_name():
-    _validate_experiment_name("validstring")
-    bytestring = b"test byte string"
-    _validate_experiment_name(bytestring.decode("utf-8"))
-    for invalid_name in ["", 12, 12.7, None, {}, []]:
+@pytest.mark.parametrize(
+    ("experiment_name", "is_good_name"),
+    [
+        ("validstring", True),
+        (b"test byte string".decode("utf-8"), True),
+        # error cases
+        ("", False),
+        (12, False),
+        (12.7, False),
+        (None, False),
+        ({}, False),
+        ([], False),
+    ],
+)
+def test_validate_experiment_name(experiment_name, is_good_name):
+    if is_good_name:
+        _validate_experiment_name(experiment_name)
+    else:
         with pytest.raises(MlflowException, match="Invalid experiment name"):
-            _validate_experiment_name(invalid_name)
+            _validate_experiment_name(experiment_name)
 
 
-def test_db_type():
-    for db_type in ["mysql", "mssql", "postgresql", "sqlite"]:
-        # should not raise an exception
+@pytest.mark.parametrize(
+    ("db_type", "is_good_type"),
+    [
+        ("mysql", True),
+        ("mssql", True),
+        ("postgresql", True),
+        ("sqlite", True),
+        # error cases
+        ("MySQL", False),
+        ("mongo", False),
+        ("cassandra", False),
+        ("sql", False),
+        ("", False),
+    ],
+)
+def test_db_type(db_type, is_good_type):
```

Comment:
I'd consider splitting the test in this case.

Reply:
How about like this?

```python
@pytest.mark.parametrize('db_type', ['mysql', 'mssql'])
def test_good_db_type(db_type):
  _validate_db_type_string(db_type)
```

Reply:
I'd name the function `test_validate_db_type_string_good`.